python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -271,11 +272,20 @@ def _build_register_command_data(
                     )
 
             if versions:
+                # Information used to construct the equivalent of Desktop "collapse rules"
+                group_name = self._get_group_name(engine)
```

Comment:
In this case when we are reading a path from a software entity, I am not sure we have enough information to do the grouping. Maybe there could be a `group` field on the software entity to help with the grouping? I'll kick off a conversation w Rob to check what he wants for this case! Outside of some hardcoded business logic magic, there doesn't seem to be an obvious simple approach to grouping here.